staging/src/k8s.io/apiserver/pkg/registry/generic/rest/proxy.go

```diff
@@ -78,6 +78,8 @@ func NewUpgradeAwareProxyHandler(location *url.URL, transport http.RoundTripper,
 
 // ServeHTTP handles the proxy request
 func (h *UpgradeAwareProxyHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	utilnet.PrepareForwardedFor(req)
```

Comment:
It seems like these modifications should be applied to newReq not req. Even if modifying req is currently safe I don't think you can rely on it remaining safe.